Framework/Algorithms/src/ConvertUnitsUsingDetectorTable.cpp

```diff
@@ -422,9 +429,8 @@ void ConvertUnitsUsingDetectorTable::convertQuickly(
     // Only do the full check if the quick one passes
     if (commonBoundaries) {
       // Calculate the new (common) X values
-      MantidVec::iterator iter;
-      for (iter = outputWS->dataX(0).begin(); iter != outputWS->dataX(0).end();
-           ++iter) {
+      for (auto iter = outputWS->mutableX(0).begin();
+           iter != outputWS->x(0).end(); ++iter) {
```

Comment:
Many of my comments on `ConvertUnits` also apply to this file, I will not repeat them here.